### PR TITLE
Re-add deleted overloads and fix namespace / extends mess.

### PR DESCRIPTION
**Pull request by Cotspheer**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
https://github.com/DefinitelyTyped/DefinitelyTyped/commit/16e5b26b3da67b6f1f693af4666db65c10596dbe
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

**Comment by NightkingCH** [comment]
Don't want to rush, but would be great to get feedback or to see this merged. We face troubles in all our projects and it would be nice to see that fixed. Don't like to cast everything to "any" :/.

File: leaflet/index.d.ts [hunk 1]
```diff
@@ -118,7 +118,7 @@ declare namespace L {
 
     export class LatLng {
         constructor(latitude: number, longitude: number, altitude?: number);
-        constructor(coords: LatLngTuple | [number, number, number] | LatLngLiteral | {lat: number, lng: number, alt?: number});
+        constructor(coords: LatLngTuple | [number, number, number] | LatLngLiteral | { lat: number, lng: number, alt?: number });
```
Comment (alejo90) [inline leaflet/index.d.ts]: Cam you remove all of these extra spaces, please.

File: leaflet/index.d.ts [hunk 2]
```diff
@@ -406,7 +406,7 @@ declare namespace L {
         getPane(name?: string): HTMLElement;
 
         // Popup methods
-        bindPopup(content: (layer: Layer) => Content | Content | Popup, options?: PopupOptions): this;
+        bindPopup(content: string | HTMLElement | Popup | ((layer: Layer) => Content), options?: PopupOptions): this;
```
Comment (alejo90) [inline leaflet/index.d.ts]: `Content` is an alias for `string | HTMLElement`, the previous definition just requires an extra set of parentheses around the function: `((layer: Layer) => Content) | Content | Popup`

File: leaflet/index.d.ts [hunk 3]
```diff
@@ -1418,34 +1418,46 @@ declare namespace L {
         className?: string;
     }
 
-    class InternalIcon extends Layer {
+    export interface IconOptions extends LayerOptions {
```
Comment (alejo90) [inline leaflet/index.d.ts]: Please remove this interface, it's defined just above.

File: leaflet/index.d.ts [hunk 4]
```diff
@@ -1418,34 +1418,46 @@ declare namespace L {
...
+        popupAnchor?: PointExpression;
+        shadowUrl?: string;
+        shadowRetinaUrl?: string;
+        shadowSize?: PointExpression;
+        shadowAnchor?: PointExpression;
+        className?: string;
+    }
+
+    class Icon extends Layer {
+        static Default: IconDefaultSettings;
```
Comment (alejo90) [inline leaflet/index.d.ts]: Except for the `InternalIcon` class (that is an artifact from when I first wrote the definitions using interfaces), the old way is correct. `Default` is not an instance, but another class and that is the reason way it was defined inside a namespace. You can verify that in the source code and in the documentation. Please revert. You are welcome to remove the `InternalIcon` class, as it is not needed anymore and it doesn't exist anyways.

File: leaflet/leaflet-tests.ts [hunk 5]
```diff
@@ -427,6 +427,23 @@ L.Control.include({});
...
+}); 
+
+// Icon > DefaultIcon > DivDicon
+// Option object should be nullable
+L.marker([2,2], { icon: icon })
+	.bindPopup("Hello Leaflet"); //string should be possible 
+
+// it should be possible to define the default path without casting
+// (<any> L.Icon.Default)
+L.Icon.Default.imagePath = "/any/valid/uri/or/path";
```
Comment (alejo90) [inline leaflet/leaflet-tests.ts]: From the documentation:

> In order to customize the default icon, just change the properties of L.Icon.Default.prototype.options (which is a set of Icon options).

As I said, `Default` is another class, not an instance.

**Review by alejo90 — CHANGES_REQUESTED** [review]
Thanks for your contribution. The `Icon` changes, however, do not reflect the way the library is written. Please remove the extra spaces that were added all over the place in the `{}`.

**Comment by alejo90** [comment]
👎 no response from author, so... I'll be submitting a fix for this and other issues this week.

**Comment by minestarks** [comment]
Looks like this now conflicts with another version of the fix already checked in. I'll close this PR. Please submit a new PR for additional changes. Thank you.

**Pull request closed without merging**